filter/set.go

```diff
@@ -0,0 +1,67 @@
+package filter
+
+import (
+	"sync"
+)
+
+// set is a Filter that contains 2 or more Filters
+type set struct {
+	filters []Filter
+}
+
+// NewSet returns a set of Filters
+func NewSet(filters ...Filter) Filter {
+	return &set{filters: filters}
+}
+
+// Run against an input and output channel until the input channel is closed.
+// The output channel is closed on return.
+func (s *set) Run(in <-chan int, out chan<- int) {
+	var wg sync.WaitGroup
+	wg.Add(len(s.filters))
+
+	if len(s.filters) <= 1 {
```

Comment:
Why not check and add this before the first `wg.Add`, and then follow the normal flow?